### PR TITLE
Add more tests for FilePackage.Headers

### PR DESCRIPTION
**Pull request by victornoel**
This is for #155.

In the end I didn't use parametrized tests as proposed by the todo because it was very complex for nothing (and difficulty maintainable I believe)

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#187](https://github.com/artipie/rpm-adapter/issues/187) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/c33a1d0d-2ba3-473b-a567-d9ffdc1d9c3d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/rpm/pkg/FilePackageHeadersTest.java [hunk 1]
```diff
@@ -35,23 +36,87 @@
...
+            ).header(tag).asString(expected),
+            new IsEqual<>(expected)
+        );
+    }
+
+    @Test
+    public void parseMissingIntHeader(@TempDir final Path unused) {
+        final Header.HeaderTag tag = Header.HeaderTag.EPOCH;
+        final int expected = 0;
+        final Header header = new Header();
```
Comment (carlosmiranda) [inline src/test/java/com/artipie/rpm/pkg/FilePackageHeadersTest.java:115]: @victornoel let's inline this variable.

File: src/test/java/com/artipie/rpm/pkg/FilePackageHeadersTest.java [hunk 2]
```diff
@@ -35,23 +36,87 @@
...
+            new FilePackage.Headers(
+                header, unused, Digest.SHA256
+            ).header(tag).asString(expected),
+            new IsEqual<>(expected)
+        );
+    }
+
+    @Test
+    public void parseMissingIntHeader(@TempDir final Path unused) {
+        final Header.HeaderTag tag = Header.HeaderTag.EPOCH;
```
Comment (carlosmiranda) [inline src/test/java/com/artipie/rpm/pkg/FilePackageHeadersTest.java:113]: @victornoel let's inline this variable.

File: src/test/java/com/artipie/rpm/pkg/FilePackageHeadersTest.java [hunk 3]
```diff
@@ -35,23 +36,87 @@
...
+            ).header(tag).asInts(),
+            new IsEqual<>(expected)
+        );
+    }
+
+    @Test
+    public void parseMissingStringHeader(@TempDir final Path unused) {
+        final Header.HeaderTag tag = Header.HeaderTag.NAME;
+        final String expected = "default string value";
+        final Header header = new Header();
```
Comment (carlosmiranda) [inline src/test/java/com/artipie/rpm/pkg/FilePackageHeadersTest.java:102]: @victornoel let's inline this variable.

File: src/test/java/com/artipie/rpm/pkg/FilePackageHeadersTest.java [hunk 4]
```diff
@@ -35,23 +36,87 @@
...
                 header, unused, Digest.SHA256
-            ).header(tag).asString("default string"),
+            ).header(tag).asInts(),
+            new IsEqual<>(expected)
+        );
+    }
+
+    @Test
+    public void parseMissingStringHeader(@TempDir final Path unused) {
+        final Header.HeaderTag tag = Header.HeaderTag.NAME;
```
Comment (carlosmiranda) [inline src/test/java/com/artipie/rpm/pkg/FilePackageHeadersTest.java:100]: @victornoel let's inline this variable.

**Review by carlosmiranda — CHANGES_REQUESTED** [review]
@victornoel please see comments

**Comment by victornoel** [comment]
@carlosmiranda it is done, thank you

**Review by carlosmiranda — APPROVED** [review]
@victornoel thanks. @g4s8 this is good to merge.

**Review by g4s8 — APPROVED** [review]

**Comment by carlosmiranda** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/rpm-adapter/issues/187#issuecomment-631761381))

@carlosmiranda This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#187](https://github.com/artipie/rpm-adapter/issues/187) is [in scope](http://www.zerocracy.com/policy.html#14) for 5days
  * The role is `REV`
  * The job is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) for [4days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will get 25% of their hourly rate for this job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @victornoel[/z](https://www.0crat.com/u/victornoel): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Frpm-adapter%23187%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@carlosmiranda please make sure during next CR, your comments would be mostly about design problems, but NOT rather cosmetic issues

**Comment by carlosmiranda** [comment]
@sereshqua confirmed

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**